src/game.h

```diff
@@ -443,8 +443,8 @@ class game
 
         void interactive_inv();
 
-        int inv_for_filter( const std::string &title, item_filter filter, const std::string &none_message = "" );
-        int inv_for_filter( const std::string &title, item_location_filter filter, const std::string &none_message = "" );
+        int inv_for_filter( const std::string &title, const item_filter &filter, const std::string &none_message = "" );
```

Comment:
Unnecessarily complex. Given that you can access all of the `item` methods from an `item_location` via indirection there's no need to provide the `item&` variant. Additionally for such a simple definition I'd avoid the typedef - I had to look up the details whereas a `std::function` would have been concise and clearer.